### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by mbevc1** [jenkinsci/prometheus-plugin#160]
### Steps to reproduce
How can we reproduce the issue
1. upgrade to 2.0.7
2. run for a while
3. Jenkins OOMs

### Expected behaviour
Should work as on 2.0.0 and Prometheus able to scrape for metrics

### Actual behaviour
Jenkins eventually runs out of memory and CPU higher than usual. Nothing specific in logs.

```
INFO: Ping failed. Terminating the channel JNLP4-connect connection from node1/:56060.
java.util.concurrent.TimeoutException: Ping started at 1591711306061 hasn't completed by 1591711551193
	at hudson.remoting.PingThread.ping(PingThread.java:133)
	at hudson.remoting.PingThread.run(PingThread.java:89)

Jun 09, 2020 2:03:55 PM hudson.triggers.Trigger checkTriggers
WARNING: Trigger hudson.triggers.TimerTrigger.run() triggered by org.jenkinsci.plugins.workflow.job.WorkflowJob@58361310[org/repo/master] spent too much time (39 sec) in its execution, other timers can be affected
Jun 09, 2020 2:15:54 PM hudson.remoting.PingThread lambda$new$0
SEVERE: Uncaught exception in PingThread Thread[Ping thread for channel hudson.remoting.Channel@138905d0:JNLP4-connect connection from node1/IP:60232,5,main]
java.lang.OutOfMemoryError: Java heap space

Jun 09, 2020 2:15:41 PM hudson.init.impl.InstallUncaughtExceptionHandler$DefaultUncaughtExceptionHandler uncaughtException
SEVERE: A thread (Periodic background build discarder thread/4083) died unexpectedly due to an uncaught exception, this may leave your Jenkins in a bad way and is usually indicative of a bug in the code.
java.lang.OutOfMemoryError: Java heap spac
```
### Server configuration

**Operating system**:
CentOS 7.8

**Jenkins Version**:
2.222.4

**Plugin Version**:
2.0.7

### Notify

@markyjackson-taulia

**Comment by mbevc1** [comment]
There is still one opened PR linked and probably after that one is sorted we need to release 2.0.8?

**Comment by pwdebruin1** [comment]
Yeah @markyjackson-taulia would be great if we can get a review on https://github.com/jenkinsci/prometheus-plugin/pull/167,  we're also seeing this issue and its been open for a while. Thanks 🙏

**Comment by mbevc1** [comment]
This is still not fixed with 2.0.8:
```
Oct 20, 2020 7:51:49 AM hudson.init.impl.InstallUncaughtExceptionHandler handleException
WARNING: Caught unhandled exception with ID cb34e4da-4856-4d56-bd16-fd4e13a5621b
java.lang.OutOfMemoryError: Java heap space

Oct 20, 2020 7:52:29 AM hudson.init.impl.InstallUncaughtExceptionHandler handleException
WARNING: Caught unhandled exception with ID 6f9e8576-0195-423c-aa0d-90ec157768f5
java.lang.OutOfMemoryError: Java heap space

Oct 20, 2020 7:59:35 AM hudson.model.AsyncPeriodicWork doRun
INFO: Periodic background build discarder thread is still running. Execution aborted.
Oct 20, 2020 8:03:31 AM org.jenkinsci.plugins.workflow.job.WorkflowRun finish
```
